integrations/tensorflow/iree_tf_compiler/TFL/test/retain_call_once_funcs.mlir

```diff
@@ -0,0 +1,21 @@
+// RUN: iree-opt-tflite -allow-unregistered-dialect -split-input-file -pass-pipeline='iree-tflite-retain-call-once-funcs' %s | IreeFileCheck %s
+
+// CHECK-LABEL: module {
+module {
+  // CHECK-LABEL: @main
+  func @main(%arg0: tensor<16x16xf32>) -> (tensor<16x16xf32>) {
+    // CHECK: "tfl.call_once"() {session_init_function = "NoOp", session_init_function_symbol = @NoOp} : () -> ()
+    "tfl.call_once"() {session_init_function = "NoOp"} : () -> ()
+    %0 = "tfl.var_handle"() {container = "", shared_name = "Variable"} : () -> tensor<*x!tf_type.resource>
+    %1 = "tfl.read_variable"(%0) : (tensor<*x!tf_type.resource>) -> tensor<16x16xf32>
+    %2 = tfl.add %1, %arg0 {fused_activation_function = "NONE"} : tensor<16x16xf32>
+    "tfl.assign_variable"(%0, %2) : (tensor<*x!tf_type.resource>, tensor<16x16xf32>) -> ()
+    return %2 : tensor<16x16xf32>
+  }
+  func private @NoOp() {
+    %0 = "tfl.var_handle"() {container = "", shared_name = "Variable"} : () -> tensor<*x!tf_type.resource>
+    %1 = "tfl.pseudo_const"() {value = dense<0.000000e+00> : tensor<16x16xf32>} : () -> tensor<16x16xf32>
+    "tfl.assign_variable"(%0, %1) : (tensor<*x!tf_type.resource>, tensor<16x16xf32>) -> ()
+    return
```

Comment:
Are these other ops necessary for the test? Should you be checking for them with `CHECK` or `CHECK-NOT`? I think from "Guarantees that functions used by tfl.call_once are retained" that you should be checking for the `@NoOp` function to remain, maybe after running DCE too?

Reply:
We could check the replacements in them however they are more a vestigial part of the state behavior. This "init op" idea means that technically they need to be replaced however once we use correct global variable syntax, they are no longer necessary. I wanted to avoid cluttering more checks than the core objectives.